Realm/Realm/Configurations/SyncConfiguration.cs

```diff
@@ -137,13 +139,15 @@ internal override Realm CreateRealm(RealmSchema schema)
         internal override async Task<Realm> CreateRealmAsync(RealmSchema schema, CancellationToken cancellationToken)
         {
             var configuration = CreateConfiguration();
+            var syncConfiguration = CreateSyncConfiguration();
+            syncConfiguration.schema_mode = ObjectClasses == null ? SchemaMode.AdditiveDiscovered : SchemaMode.AdditiveExplicit;
```

Comment:
This can be moved to `CreateSyncConfiguration` instead of repeating it everywhere.